src/pages/AddProduct.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState } from "react";
+import { addProduct } from "../api/products";
+
+const AddProduct = () => {
+  const [product, setProduct] = useState({
+    name: "",
+    price: 0,
+    description: "",
+    stock: 0,
+    file: null,
+  });
+
+  const handleChange = (key: string, value: any) => {
+    setProduct({ ...product, [key]: value });
+    console.log(product, "HERE");
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (!product) return;
```

Comment:
With the default state of the useState this should never happen. Maybe you can make a utility function `checkRequiredProductFields` or something like that to check if the product has all the required data. Show a nice error message with what fields still need to be filled in

Reply:
Yes, sure. This is just a very basic form to upload products from the front instead of adding them to the db directly.